definitions/procedures/packages/update.rb

```diff
@@ -11,7 +11,7 @@ class Update < ForemanMaintain::Procedure
 
     def run
       assumeyes_val = @assumeyes.nil? ? assumeyes? : @assumeyes
-      package_manager.clean_cache(:assumeyes => assumeyes_val)
+      package_manager.clean_cache(:assumeyes => assumeyes_val) if download_only?
```

Comment:
I think gating this on `download_only` being true is wrong.

We use `Procedures::Packages::Update` in other places, that don't use `download_only` and thus would not get cache refreshes anymore:
https://github.com/theforeman/foreman_maintain/blob/fa2e3cb52fc6d224124bd9d3e9d1c888578625a8/definitions/scenarios/self_upgrade.rb#L67-L87
https://github.com/theforeman/foreman_maintain/blob/fa2e3cb52fc6d224124bd9d3e9d1c888578625a8/definitions/scenarios/upgrade_to_foreman_nightly.rb#L59

I think we need an own param that controls the cache cleanup, which defaults to `true` and then the "second" runs (after download_only) set it explicitly to `false`